src/compile/axis/properties.ts

```diff
@@ -167,58 +169,111 @@ export function getLabelAngle(
   }
 }
 
+export function normalizeAngleExpr(angle: SignalRef) {
+  return `(((${angle.signal} % 360) + 360) % 360)`;
+}
+
 export function defaultLabelBaseline(
-  angle: number,
-  axisOrient: AxisOrient,
+  angle: number | SignalRef,
+  orient: AxisOrient | SignalRef,
   channel?: 'x' | 'y',
   alwaysIncludeMiddle?: boolean
 ) {
-  channel = channel || (axisOrient === 'top' || axisOrient === 'bottom' ? 'x' : 'y');
+  channel = channel || (orient === 'top' || orient === 'bottom' ? 'x' : 'y');
 
   if (angle !== undefined) {
     if (channel === 'x') {
-      return (45 < angle && angle < 135) || (225 < angle && angle < 315)
-        ? 'middle'
-        : (angle <= 45 || 315 <= angle) === (axisOrient === 'top')
-        ? 'bottom'
-        : 'top';
+      if (isSignalRef(angle)) {
+        const a = normalizeAngleExpr(angle);
+        const orientIsTop = isSignalRef(orient) ? `(${orient.signal} === "top")` : orient === 'top';
+        return {
+          signal:
+            `(45 < ${a} && ${a} < 135) || (225 < ${a} && ${a} < 315) ? "middle" :` +
+            `(${a} <= 45 || 315 <= ${a}) === ${orientIsTop} ? "bottom" : "top"`
+        };
+      }
+
+      if ((45 < angle && angle < 135) || (225 < angle && angle < 315)) {
+        return 'middle';
+      }
+
+      if (isSignalRef(orient)) {
```

Comment:
This looks unreachable since `isSignalRef(orient)` implies `channel === 'y'`. (I think we need a signal check on the `channel` assignment line).

Reply:
Why does `isSignalRef(orient)` implies `channel === 'y'`?

Reply:
FWIW, this is *not* unreachable -- look at the test coverage.

Reply:
I'll remove the `channel` inference code so it's clear.